runtime/lua/vim/_meta/vvars_extra.lua

```diff
@@ -63,8 +63,10 @@ error('Cannot require a meta file')
 --- complete item selected.
 --- @field height? integer
 --- @field width? integer Height of popup menu on |CompleteChanged|
---- @field row? integer Width of popup menu on |CompleteChanged|
---- Col count of popup menu on |CompleteChanged|, relative to screen.
+--- @field row? integer For |CompleteChanged|: width of popup menu. For |TermRequest|: current
+--- cursor row when escape sequence was received.
+--- For |CompleteChanged|: col count of popup menu, relative to screen. For |TermRequest|: current
+--- cursor column when escape sequence was received.
```

Comment:
Typically `:help v:event` should not need to enumerate every event for a field; rather the field should describe is general purpose, then each help doc for each event mentions the `v:event` fields that it sets.

IOW: `:help TermRequest` should mention these v:event fields. 

Also, `:help TermRequest` already mentions `:help event-data` but I see that it sets `args.data` to a string. I think in general we shouldn't be shy about adding more fields to `args` (and those fields should ideally match `v:event` where possible).

Reply:
`args.data` has been the place where we put arbitrary event data, reserving the top level table for well-known (and documented) fields like "buf", "file", etc. If we start putting arbitrary data in the top-level then it (1) risks conflicts with the well-known fields (users can add arbitrary data with `nvim_exec_autocmds`), and (2) makes the `data` sub-field meaningless (or, at best, confusing).

I don't think we should add new event data to the top-level field, but we could consider making a breaking change to the TermRequest autocommand to make `args.data` a table instead. Then we would have `args.data.payload` for the actual escape sequence, and `args.data.row` and `args.data.col`.

Reply:
>  but we could consider making a breaking change to the TermRequest autocommand to make `args.data` a table instead. Then we would have `args.data.payload` for the actual escape sequence, and `args.data.row` and `args.data.col`.

SGTM